### ISSUE TITLE
Nested layout views and attribute access

### ISSUE THREAD

**Issue by cdlm** [nanoc/nanoc#628]
Nanoc seems to be nesting views, is it intended?
```
[1] pry(#<ByExtension>)> assigns[:layout]
=> #<Nanoc::LayoutView:0x007fd3d391fe70
 @document=
  #<Nanoc::LayoutView:0x007fd3d3915f88
   @document=<Nanoc::Int::Layout identifier="/partials/markdown_links.erb">>>
```

Anyway, nested view work fine for accessors:
```
[2] pry(#<ByExtension>)> assigns[:layout].identifier
=> <Nanoc::Identifier type=full "/partials/markdown_links.erb">
[3] pry(#<ByExtension>)> assigns[:layout].identifier.ext
=> "erb"
```

The layout inside has some attributes:
```
[4] pry(#<ByExtension>)> assigns[:layout].unwrap.unwrap.attributes
=> {:filename=>"layouts/partials/markdown_links.erb",
 :content_filename=>"layouts/partials/markdown_links.erb",
 :meta_filename=>nil,
 :extension=>"erb",
 :mtime=>2014-11-23 20:30:58 +0100}
```

But accessing them via `[]` fails:
```
[5] pry(#<ByExtension>)> assigns[:layout][:extension]
NoMethodError: undefined method `attributes' for #<Nanoc::LayoutView:0x007fd3d3915f88>
from /Users/damien/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/nanoc-5231a7247477/lib/nanoc/base/views/mixins/document.rb:37:in `[]'
```

**Comment by ddfreyne** [comment]
Ah, that’s not quite right. Will fix.

**Comment by ddfreyne** [comment]
Potential fix in #631.

**Comment by ddfreyne** [comment]
@cdlm Will close not, but please double-check if the issue is truly solved for you.
